### PR TITLE
Add type hints for layer gradient SHAP

### PR DESCRIPTION
**Pull request by NarineK**
Adding type hints in layer gradient SHAP and related tests

### CODE REVIEW

File: captum/attr/_core/layer/layer_gradient_shap.py [hunk 1]
```diff
@@ -39,6 +50,37 @@ def __init__(self, forward_func, layer, device_ids=None):
         LayerAttribution.__init__(self, forward_func, layer, device_ids)
         GradientShap.__init__(self, forward_func)
 
+    @typing.overload
+    def attribute(
+        self,
+        inputs: TensorOrTupleOfTensors,
+        baselines: TensorOrTupleOfTensors,
```
Comment (NarineK) [inline captum/attr/_core/layer/layer_gradient_shap.py]: need to add callable here

File: tests/attr/layer/test_layer_gradient_shap.py [hunk 2]
```diff
@@ -112,33 +117,33 @@ def test_basic_multi_input(self):
...
-            self, attrs, expected, delta=0.005,
-        )
+        assertTensorTuplesAlmostEqual(self, attrs, expected, delta=0.005)
         if expected_delta is None:
             _assert_attribution_delta(self, inputs, attrs, n_samples, delta, True)
         else:
-            for delta_i, expected_delta_i in zip(delta, expected_delta):
-                self.assertAlmostEqual(delta_i, expected_delta_i, delta=0.01)
+            for delta_i, expected_delta_i in zip(
+                torch.unbind(delta), torch.unbind(expected_delta)
```
Comment (vivekmig) [inline tests/attr/layer/test_layer_gradient_shap.py]: Do we need the unbind calls here? I think this should type hint fine without it if testing with torch 1.4.

File: tests/attr/layer/test_layer_gradient_shap.py [hunk 3]
```diff
@@ -112,33 +117,33 @@ def test_basic_multi_input(self):
...
-        n_samples=5,
-        attribute_to_layer_input=False,
-        add_args=None,
-    ):
+        model: Module,
+        layer: Module,
+        inputs: TensorOrTupleOfTensors,
+        baselines: Union[TensorOrTupleOfTensors, Callable],
+        target: Union[int, Tuple[int, ...], Tensor, List[Tuple[int, ...]]],
+        expected: Union[Tensor, Tuple[Tensor, ...], List, Tuple[List, ...]],
```
Comment (vivekmig) [inline tests/attr/layer/test_layer_gradient_shap.py]: nit: Could make the List hint more specific? (List[float] or List[List[float]]?)

File: tests/attr/layer/test_layer_gradient_shap.py (hunk 2 above)
Comment (NarineK) [reply tests/attr/layer/test_layer_gradient_shap.py]: Cool, yeah, it works w/1.4 without unbinding.

File: tests/attr/layer/test_layer_gradient_shap.py (hunk 3 above)
Comment (NarineK) [reply tests/attr/layer/test_layer_gradient_shap.py]: It makes it a little more lengthy but added it

**Pull request closed without merging**